docs/openshift/deploy.md

```diff
@@ -0,0 +1,102 @@
+* A location to provision the cluster e.g. `eastus`.
+
+```
+$ az account list -o table
+Name                                             CloudName    SubscriptionId                        State    IsDefault
+-----------------------------------------------  -----------  ------------------------------------  -------  -----------
+Acme Corp                                        AzureCloud   5eca53b6-18b4-4d9b-a4d4-a45a1ff367c8  Enabled  True
+```
+
+## Deploy
+
+For this example, the subscription id is `5eca53b6-18b4-4d9b-a4d4-a45a1ff367c8`, the DNS prefix is `openshift-red`, and location is `eastus`.
+
+Run `acs-engine deploy` with the appropriate arguments:
+
+```
+$ acs-engine deploy --subscription-id 5eca53b6-18b4-4d9b-a4d4-a45a1ff367c8 \
+    --dns-prefix openshift-red --location eastus \
+    --api-model examples/openshift.json
+
+INFO[0034] Starting ARM Deployment (openshift-red-1843927849). This will take some time...
+INFO[0393] Finished ARM Deployment (openshift-red-1843927849).
+```
+
+`acs-engine` will output Azure Resource Manager (ARM) templates, SSH keys and a node configuration in `_output/openshift-red` directory:
+
+   * `_output/openshift-red/apimodel.json`
+   * `_output/openshift-red/azuredeploy.json`
+   * `_output/openshift-red/azuredeploy.parameters.json`
+   * `_output/openshift-red/master.tar.gz`
+   * `_output/openshift-red/node.tar.gz`
+
+Administrative note: By default, the directory where acs-engine stores cluster
+configuration (`_output/openshift-red` above) won't be overwritten as a result
+of subsequent attempts to deploy a cluster using the same `--dns-prefix`) To
+re-use the same resource group name repeatedly, include the `--force-overwrite`
+command line option with your `acs-engine deploy` command. On a related note,
+include an `--auto-suffix` option to append a randomly generated suffix to the
+dns-prefix to form the resource group name, for example if your workflow
+requires a common prefix across multiple cluster deployments. Using the
+`--auto-suffix` pattern appends a compressed timestamp to ensure a unique
+cluster name (and thus ensure that each deployment's configuration artifacts
+will be stored locally under a discrete `_output/<resource-group-name>/`
+directory).
+
+<a href="#the-long-way"></a>
+
+## ACS Engine the Long Way
+
+### Step 1: Generate an SSH Key
+
+In addition to using OpenShift APIs to interact with the clusters, cluster operators may access the master machine using SSH.
+
+If you don't have an SSH key [cluster operators may generate a new one](../ssh.md#ssh-key-generation).
+
+### Step 2: Create a Service Principal
+
+OpenShift clusters have integrated support for various cloud providers as core functionality. On Azure, acs-engine uses a Service Principal to interact with Azure Resource Manager (ARM). Follow the instructions to [create a new service principal](../serviceprincipal.md)
```

Comment:
The preferred route will be for acs-engine to do this for you, as long as you have the right permissions in place - because the application object will  also be used for AAD authentication, and the setup that has to be done for that is quite fiddly